### DISCUSSION TITLE
Linkerd2 helm chart installation and Rancher2 monitoring integration.

### DISCUSSION THREAD

**Discussion by hetii** [linkerd/linkerd2#5224] [Q&A]
Hello.

I use terraform+rancher2 to provision my cluster.

Rancher2 has own monitoring chart that provision grafana and prometheus instances.

My linkerd and linkerd-cni is installed via helm charts and I disable in its value file grafana and prometheus installation.

Now to reuse monitoring from rancher2 in my other deployments usually I enable `serviceMonitor` in 
`cattle-prometheus` namespace. Then the metrics was automatically scrapped by Prometheus instance.

For linkerd helm chart I don't see such possibilities, so any clue how this integration should look like?

Also do I need then install some special plugins/charts to this existing grafana?

**Comment by grampelberg** [comment]
You'll want to copy [the scrape config](https://github.com/linkerd/linkerd2/blob/main/charts/add-ons/prometheus/templates/prometheus.yaml#L97) and import the grafana dashboards. That should be it though!

**Reply by hetii** [reply]
Ok Thank you for this tip, I will check it.

**Reply by hetii** [reply]
Btw witch chart are linkerd specific that should be imported?

**Comment by hetii** [comment]
Small update from my side.
I get secret in rancher called `prometheus-cluster-monitoring-additional-scrape-configs` where rancher2 monitoring store scrape configs for istio and I add there also scrape for linkerd-proxy. 
After applying new secret in prometheus target i see `linkerd-proxy (8/8 up)`.
In service discovery tab I see also: linkerd-proxy (8/79 active targets)

Then when I try search some metrics based on `https://linkerd.io/2/reference/proxy-metrics` I get some statistic but
but `linkerd stat deployments -A` show stats like that:

	NAMESPACE                 NAME                      MESHED   SUCCESS   RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99   TCP_CONN
	linkerd                   linkerd-controller           1/1         -     -             -             -             -          -
	linkerd                   linkerd-destination          1/1         -     -             -             -             -          -
	linkerd                   linkerd-identity             1/1         -     -             -             -             -          -
	linkerd                   linkerd-proxy-injector       1/1         -     -             -             -             -          -
	linkerd                   linkerd-sp-validator         1/1         -     -             -             -             -          -
	linkerd                   linkerd-tap                  1/1         -     -             -             -             -          -
	linkerd                   linkerd-web                  1/1         -     -             -             -             -          -

So what I missing there and how linkerd cli know with instance of prometheus it should use to display properly it's stats?
